src/main/webapp/app/core/course/shared/course-base-container/course-base-container.component.ts

```diff
@@ -0,0 +1,343 @@
+        this.vcSubscription?.unsubscribe();
+        this.subscription?.unsubscribe();
+        this.profileSubscription?.unsubscribe();
+        this.closeSidebarEventSubscription?.unsubscribe();
+        this.openSidebarEventSubscription?.unsubscribe();
+        this.ltiSubscription?.unsubscribe();
+        this.loadCourseSubscription?.unsubscribe();
+        this.dashboardSubscription?.unsubscribe();
+        this.ngUnsubscribe.next();
+        this.ngUnsubscribe.complete();
+    }
+
+    /** Initialize courses attribute by retrieving all courses from the server */
+    async updateRecentlyAccessedCourses() {
+        this.dashboardSubscription = this.courseManagementService.findAllForDropdown().subscribe({
+            next: (res: HttpResponse<Course[]>) => {
+                if (res.body) {
+                    let courses: Course[] = [];
+                    res.body?.forEach((course) => {
+                        courses.push(course);
+                    });
+                    courses = sortCourses(courses);
+                    if (courses.length > this.MIN_DISPLAYED_COURSES) {
+                        const lastAccessedCourseIds = this.courseAccessStorageService.getLastAccessedCourses(CourseAccessStorageService.STORAGE_KEY_DROPDOWN);
+                        courses = courses.filter((course) => lastAccessedCourseIds.includes(course.id!));
+                    }
+                    courses = courses.filter((course) => course.id !== this.courseId());
+                    this.courses.set(courses);
+                }
+            },
+        });
+    }
+
+    /**
+     * Accepts a component reference of the subcomponent rendered based on the current route.
+     * If it provides a controlsConfiguration, we try to render the controls component
+     */
+    onSubRouteActivate(componentRef: any) {
+        this.getPageTitle();
+        const urlSegments = this.router.url.split('?')[0].split('/');
+        this.communicationRouteLoaded.set(urlSegments.length > 3 && urlSegments[3] === 'communication');
+
+        this.hasSidebar.set(this.getHasSidebar());
+        this.setupConversationService();
+
+        if (componentRef.controlConfiguration) {
+            const provider = componentRef;
+            this.controlConfiguration.set(provider.controlConfiguration);
+
+            this.controlsSubscription = provider.controlConfiguration?.subject?.subscribe(async (controls: TemplateRef<any>) => {
+                this.controls.set(controls);
+                this.tryRenderControls();
+                await firstValueFrom(provider.controlsRendered);
+                this.tryRenderControls();
+            });
+        }
+
+        // Handle component specific activations in child classes
+        this.handleComponentActivation(componentRef);
+
+        // Since we change the pageTitle + might be pulling data upwards during a render cycle, we need to re-run change detection
+        this.changeDetectorRef.detectChanges();
```

Comment:
```suggestion
            });
        }

        this.handleComponentActivation(componentRef);

        // Since we change the pageTitle + might be pulling data upwards during a render cycle, we need to re-run change detection
        this.changeDetectorRef.detectChanges();
```